src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -466,11 +467,278 @@ type Moon {
+                        ""graphql"": {
+                          ""enabled"": " + gQLEntityEnabled + @",
+                          ""type"": {
+                            ""singular"": """ + gQLEntitySingular + @""",
+                            ""plural"": """ + gQLEntityPlural + @"""
+                          }
+                        },
+                        ""rest"": {
+                          ""enabled"": " + restEntityEnabled + @"
+                        },
+                        ""permissions"": [
+                        {
+                          ""role"": ""anonymous"",
+                          ""actions"": [
+                            {
+                              ""action"": ""create""
+                            },
+                            {
+                              ""action"": ""read""
+                            },
+                            {
+                              ""action"": ""update""
+                            },
+                            {
+                              ""action"": ""delete""
+                            }
+                          ]
+                        },
+                        {
+                          ""role"": ""authenticated"",
+                          ""actions"": [
+                            {
+                              ""action"": ""create""
+                            },
+                            {
+                              ""action"": ""read""
+                            },
+                            {
+                              ""action"": ""update""
+                            },
+                            {
+                              ""action"": ""delete""
+                            }
+                          ]
+                        }
+                      ],
+                        ""mappings"": {
+                          """ + entityBackingColumn + @""": """ + entityExposedName + @"""
+                        }
+                      }
+                  }
+              }");
+
+        }
+
+            [TestCleanup]
         public void CleanupAfterEachTest()
         {
             TestHelper.UnsetAllDABEnvironmentVariables();
+            if (File.Exists(HOT_RELOAD_CONFIG_FILE_NAME))
+            {
+                File.Delete(HOT_RELOAD_CONFIG_FILE_NAME);
+            }
+        }
+
+        /// <summary>
+        /// Start DAB with the pre-formed config file, assert that rest and GQL
+        /// requests for the book database object returns the expected results.
+        /// Then hot-reload the config runtime paths both for rest and GQL, and
+        /// again assert that the requests return the expected results against
+        /// the book database object.
+        /// </summary>
+        /// <returns></returns>
+        [TestCategory(MSSQL_ENVIRONMENT)]
+        [TestMethod("Hot-reload runtime paths.")]
+        public async Task HotReloadConfigRuntimePathsEndToEndTest()
+        {
```

Comment:
This summary differs from the test method name which describes what you're validating here. The test is checking that a hot reloaded config changes the GraphQL and REST paths, yes? Validating these changes would ensure that old path -> 404 and new path -> 200

Reply:
Since there's only 1 path property validating that the new path works ought to also cover the old path not working, but I can add in an assert that the old path isnt valid.